cmd/manager/aggregator.go

```diff
@@ -382,14 +388,64 @@ func createResults(crClient *complianceCrClient, scan *compv1alpha1.ComplianceSc
 			"ComplianceRemediation.Namespace", crkey.Namespace)
 		remExists := getObjectIfFound(crClient, remkey, foundRemediation)
 		if remExists {
+			// If the remediation is already applied and the status of the check is compliant, only update
+			// the remediation if the payload differs. Let's not create remediations for checks that are passing
+			// needlessly and let's not trigger the remediation controller needlessly
+			if foundRemediation.Status.ApplicationState == compv1alpha1.RemediationApplied {
+				if !foundRemediation.RemediationPayloadDiffers(pr.Remediation) {
+					log.Info("Not updating passing remediation that was the same between runs", "ComplianceRemediation.Name", foundRemediation.Name)
+					continue
+				}
+
+				// Applied remediation that differs must be updated, let's set the appropriate state
+				stateUpdate = compv1alpha1.RemediationOutdated
+				log.Info("The status is now", "status", pr.Remediation.Status.ApplicationState)
+				// The old state must be kept in the outdated field so that the admin can switch to
+				// the current state at their own pace
+				foundRemediation.Spec.Current.DeepCopyInto(&pr.Remediation.Spec.Outdated)
+				// The application state must be preserved or else we'd un-apply the remediation
+				// once it's updated
+				pr.Remediation.Spec.Apply = foundRemediation.Spec.Apply
+			}
+
 			// Copy resource version and other metadata needed for update
 			foundRemediation.ObjectMeta.DeepCopyInto(&pr.Remediation.ObjectMeta)
+			// Copy the status; in case we are updating an already applied remediation
```

Comment:
Might be some missing code/TODO related to this comment?

Reply:
Copy-paste error.